### PR TITLE
Cleaned up the share view layout and fixed iOS addToHome

### PR DESCRIPTION
**Pull request by aoby**
## Old layout:

screen shot 2015-12-01 at 1 17 26 pm
### plus junk below footer:

screen shot 2015-12-01 at 1 17 40 pm
## New layout:

screen shot 2015-12-01 at 1 17 52 pm

This is similar on other apps, but without the Minecraft-specific border.

### CODE REVIEW

**Comment by aoby** [comment]
@Bjvanminnen @philbogle

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -251,6 +251,15 @@ StudioApp.prototype.init = function(config) {
     config = {};
   }
 
+  if( config.isLegacyShare ) {
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: nit `if (config` instead of `if( config`

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -251,6 +251,15 @@ StudioApp.prototype.init = function(config) {
...
+    $("body").addClass("legacy-share-view");
+    if( dom.isIOS() ) {
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: same nit

File: apps/style/common.scss [hunk 3]
```diff
@@ -644,6 +644,10 @@ input[type="radio"] {
   text-align: justify;
   -ms-text-justify: distribute-all-lines;
   text-justify: distribute-all-lines;
+
+  .legacy-share-view & {
+
+  }
```
Comment (Bjvanminnen) [inline apps/style/common.scss]: what is this accomplishing?

File: apps/src/StudioApp.js [hunk 4]
```diff
@@ -251,6 +251,15 @@ StudioApp.prototype.init = function(config) {
...
+      if (!window.navigator.standalone) {
+        addToHome.show(true);
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js:258]: Have we tested that addToHome works now?

**Comment by Bjvanminnen** [comment]
Does flappy share page (i.e. https://studio.code.org/c/31564065) end up with the "Make Your Own Flappy Game" with this change? Not sure whether or not we want it to these days. The current behavior on prod here is clearly broken for small screens:
image

File: apps/src/StudioApp.js (hunk 4 above)
Comment (aoby) [inline apps/src/StudioApp.js:258]: Yes, on the ipad last week. Note that was only from my localhost so it was not behind CloudFront

File: apps/src/StudioApp.js (hunk 1 above)
Comment (aoby) [inline apps/src/StudioApp.js]: fixed

File: apps/src/StudioApp.js (hunk 2 above)
Comment (aoby) [inline apps/src/StudioApp.js]: fixed

File: apps/style/common.scss (hunk 3 above)
Comment (aoby) [inline apps/style/common.scss]: Nothing. That was a mistake, leftover from an earlier prototype. I realized I could hide that area by not rendering the reference area. Removed

**Comment by aoby** [comment]
Flappy share view:

screen shot 2015-12-01 at 1 39 43 pm

File: apps/src/StudioApp.js [hunk 5]
```diff
@@ -1579,8 +1588,8 @@ StudioApp.prototype.handleHideSource_ = function (options) {
       $('body').append(wireframeSendToPhone);
     }
     document.body.style.backgroundColor = '#202B34';
-  // For share page on mobile, do not show this part.
-  } else if (!options.embed && !(this.share && dom.isMobile())) {
+  // For share page, do not show this part.
+  } else if (!options.embed && !this.share) {
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js:1592]: Do you know if we ever still hit this block? i.e. is there a scenario in which we're calling handleHideSource_ but both embed and share are false?

**Comment by Bjvanminnen** [comment]
Presumably we're okay with not having the "Make your own flappy game" upsell anymore @alicesteinglass. I think this is probably fine.

File: apps/src/StudioApp.js (hunk 5 above)
Comment (aoby) [inline apps/src/StudioApp.js:1592]: I don't know. I don't have enough context on where this is called. Pointers?

Comment (Bjvanminnen) [inline apps/src/StudioApp.js:1592]: My hypothesis is that this is dead code now, but I only say that with like 70% confidence.

**Pull request merged**